### PR TITLE
Feature/#8024 dcc validation/#8097 http countries

### PR DESCRIPTION
**Pull request by pascal-brause**
## Description
Introduces the HTTP call to load the available lists of onboarded countries for the dcc validation. Returns our well known and loved PackageDownloadResponse, which can be later verified and extracted to get the real country codes list by the validationService.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-8097

## Screenshots
Imagine a cool picture of an http GET call.

### CODE REVIEW

**Review by KaiTeuber — APPROVED** [review]

**Review by 30mar — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift [hunk 1]
```diff
@@ -614,7 +614,51 @@ final class HTTPClient: Client {
 			}
 		})
 	}
-
+	
+	func getDccOnboardedCountries(
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift:618]: getDCCOnboardedCountries

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift [hunk 2]
```diff
@@ -614,7 +614,51 @@ final class HTTPClient: Client {
...
+					}
+					let etag = response.httpResponse.value(forCaseInsensitiveHeaderField: "ETag")
+					let packageDownloadResponse = PackageDownloadResponse(package: sapPackage, etag: etag)
+					Log.info("Successfully got list of dcc onboarded countries", log: .api)
+					completion(.success(packageDownloadResponse))
+				case 304:
+					Log.info("Content was not modified - 304.", log: .api)
+					completion(.failure(.notModified))
+				default:
+					Log.error("Generell server error.", log: .api)
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift:653]: General

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+GetDccOnboardedCountries.swift [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+			responseData: Data())
+		let client = HTTPClient.makeWith(mock: stack)
+		let expectation = self.expectation(description: "completion handler is called with notModified failure")
+		var failure: URLSession.Response.Failure?
+		
+		// WHEN
+		client.getDccOnboardedCountries(completion: { result in
+			switch result {
+			case .success:
+				XCTFail("This test should not success.")
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+GetDccOnboardedCountries.swift:55]: succeed

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+GetDccOnboardedCountries.swift [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+GetDccOnboardedCountries.swift:80]: succeed

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+GetDccOnboardedCountries.swift [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+			responseData: Data())
+		let client = HTTPClient.makeWith(mock: stack)
+		let expectation = self.expectation(description: "completion handler is called with serverError failure")
+		var failure: URLSession.Response.Failure?
+		
+		// WHEN
+		client.getDccOnboardedCountries(completion: { result in
+			switch result {
+			case .success:
+				XCTFail("This test should not success.")
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+GetDccOnboardedCountries.swift:105]: succeed

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+GetDccOnboardedCountries.swift [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+			responseData: nil)
+		let client = HTTPClient.makeWith(mock: stack)
+		let expectation = self.expectation(description: "completion handler is called with invalidResponse failure")
+		var failure: URLSession.Response.Failure?
+		
+		// WHEN
+		client.getDccOnboardedCountries(completion: { result in
+			switch result {
+			case .success:
+				XCTFail("This test should not success.")
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+GetDccOnboardedCountries.swift:130]: succeed

File: src/xcode/ENA/ENA/Source/Client/__tests__/Mocks/ClientMock.swift [hunk 7]
```diff
@@ -64,6 +64,7 @@ final class ClientMock {
 	var onTraceWarningDownload: ((String, Int, @escaping TraceWarningPackageDownloadCompletionHandler) -> Void)?
 	var onDCCRegisterPublicKey: ((Bool, String, String, @escaping DCCRegistrationCompletionHandler) -> Void)?
 	var onGetDigitalCovid19Certificate: ((String, Bool, @escaping DigitalCovid19CertificateCompletionHandler) -> Void)?
+	var onGetDccOnboardedCountries: ((Bool, @escaping DCCOnboardedCountriesCompletionHandler) -> Void)?
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Client/__tests__/Mocks/ClientMock.swift:67]: onGetDCCOnboardedCountries

File: src/xcode/ENA/ENA/Source/Client/__tests__/Mocks/ClientMock.swift [hunk 8]
```diff
@@ -275,13 +275,24 @@ extension ClientMock: Client {
...
-
 	) {
 		guard let onGetDigitalCovid19Certificate = self.onGetDigitalCovid19Certificate else {
 			completion(.success((DCCResponse(dek: "dataEncryptionKey", dcc: "coseObject"))))
 			return
 		}
 		onGetDigitalCovid19Certificate(token, isFake, completion)
 	}
+	
+	func getDccOnboardedCountries(
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Client/__tests__/Mocks/ClientMock.swift:286]: getDCCOnboardedCountries

**Comment by naveeddotio** [comment]
Small thing. Since DCC is an abbreviation of digital covid certificate, I think we should go with DCC rather than Dcc for consistency purposes, similar to how you are have used in completion handlers like DCCRegistrationCompletionHandler and DCCOnboardedCountriesCompletionHandler

**Comment by pascal-brause** [comment]
> Small thing. Since DCC is an abbreviation of digital covid certificate, I think we should go with DCC rather than Dcc for consistency purposes, similar to how you are have used in completion handlers like DCCRegistrationCompletionHandler and DCCOnboardedCountriesCompletionHandler

Thats okay for me!

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**